packages/cli/src/ActiveExecutions.ts

```diff
@@ -186,11 +196,28 @@ export class ActiveExecutions {
 		return this.getExecution(executionId).status;
 	}
 
+	async cancelEnqueuedExecutionsWithResponsePromises() {
+		const executionIds = Object.entries(this.activeExecutions)
+			.filter(([_, execution]) => execution.status === 'new' && execution.responsePromise)
+			.map(([executionId, _]) => executionId);
+
+		if (executionIds.length === 0) return;
+
+		await this.executionRepository.cancelMany(executionIds);
+
+		this.logger.info('Canceled enqueued executions with response promises', { executionIds });
+	}
+
 	/** Wait for all active executions to finish */
 	async shutdown(cancelAll = false) {
 		let executionIds = Object.keys(this.activeExecutions);
 
 		if (cancelAll) {
+			if (config.getEnv('executions.mode') === 'regular') {
+				this.concurrencyControl.removeAll();
+				await this.cancelEnqueuedExecutionsWithResponsePromises();
```

Comment:
I'd suggest we call this from within the `ConcurrencyControl.removeAll` method, so it's all centralized wdyt?

Reply:
[622eb47](https://github.com/n8n-io/n8n/pull/9453/commits/622eb47161944752b054644f1655e1d919e8932e)